src/util-file.h

```diff
@@ -85,10 +85,13 @@ typedef struct File_ {
 #endif
     struct File_ *next;
     SCMd5 *md5_ctx;
+    uint32_t md5_num_bytes;
```

Comment:
I'm not too excited about having to grow this structure like this. Is there another way we can do this?

If not, at least the in-memory layout needs to be improved by avoiding holes in the memory layout (hint: check `pahole`)

Reply:
Couldn't find a reasonable way to remove these without removing functionality (ie., only using one hash function, not allowing multiple hashing functions).
SHA1 is only 20 bytes, so there's room to squeeze one in there; the others I added after, so they were at least co-located.